docs/contributing/3.0-development/plugin-layout.rst

```diff
@@ -84,3 +84,12 @@ of a plugin app, and so on.
 
 This matching of module names is required for the Pulp Platform to be able to auto-discover
 plugin components, particularly for both models and viewsets.
+
+
+Packaging
+---------
+
+The Plugin API is available from PyPI as pulpcore-plugin. A plugin writer needs to specify the
+version of pulpcore-plugin their plugin is dependent on. A plugin writer does not need to be
+concerned which version of pulpcore would work with their plugin since pulpcore-plugin will
+resolve the pulpcore dependency.
```

Comment:
Yeah maybe the release notes can articulate the version of platform that the plugin API version is compatible with. It's a release by release thing so that probably is the place for it.